### PR TITLE
Fix function interface for PorousDelftEgg and PorousModifiedCamClay

### PR DESCRIPTION
**Pull request by jhuang2601**
This PR 
- [x] fix function interface for `PorousDelftEgg` and `PorousModifiedCamClay`
- [x] add smoke tests for `PorousDelftEgg` and `PorousModifiedCamClay` to the integratedTests and check the run on Pecan.   
       [GEOSX/integratedTests#222](https://github.com/GEOSX/integratedTests/pull/222)      
- [x] prepare documents for these two models

### CODE REVIEW

**Review by herve-gross — APPROVED** [review]

File: inputFiles/poromechanics/PoroDelftEggWellbore_base.xml [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+<!-- SPHINX_WELLBORE_TABLE -->
+<Functions>
+  <TableFunction 
+    name="timeFunction"
+    inputVarNames="{time}"
+    coordinates="{0.0, 0.1, 1e6}"
+    values="{0.0, 1.0, 1.0}" 
+  />
+</Functions>
+<!-- SPHINX_WELLBORE_TABLE_END -->
```
Comment (joshua-white) [inline inputFiles/poromechanics/PoroDelftEggWellbore_base.xml:11]: Do you need these tags?  Do you plan to document this file with Sphinx?

File: inputFiles/poromechanics/PoroDelftEggWellbore_base.xml [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    defaultRecompressionIndex="0.001"
+  />
+
+  <CompressibleSinglePhaseFluid 
+    name="water"
+    defaultDensity="1000"
+    defaultViscosity="0.001"
+    referencePressure="0e6"
+    referenceDensity="1000"
+    compressibility="2.09028227021e-10"
```
Comment (joshua-white) [inline inputFiles/poromechanics/PoroDelftEggWellbore_base.xml:40]: Are 11 digits of precision enough? 😃 Maybe we can stick to ~4.

**Review by joshua-white — APPROVED** [review]
Looks great to me.  Let's just keep an eye on the build time to see if it is substantially larger than the current develop version.  Hopefully not, but every new model combination adds up.

File: inputFiles/poromechanics/PoroDelftEggWellbore_base.xml (hunk 1 above)
Comment (jhuang2601) [reply inputFiles/poromechanics/PoroDelftEggWellbore_base.xml]: Not for now and remove these tags

File: inputFiles/poromechanics/PoroDelftEggWellbore_base.xml (hunk 2 above)
Comment (jhuang2601) [reply inputFiles/poromechanics/PoroDelftEggWellbore_base.xml]: Right, no need for such precision

**Comment by jhuang2601** [comment]
> Looks great to me. Let's just keep an eye on the build time to see if it is substantially larger than the current develop version. Hopefully not, but every new model combination adds up.

It seems that there is no big change of the building time

**Pull request merged**